### ISSUE TITLE
rtrim sometimes breaks realText

### ISSUE THREAD

**Issue by vadimonus** [fzaninotto/Faker#1378]
in Text.php - appendEnd is used rtim function, wich is not utf safe. This sometimes causes error in next case.
Sometimaes generated text finished with russian "р" (unicode 0xD180)
As far as rtrim is singlebyte, it treats ",— " not as ","(0x2C) + "—"(0xE28094) + " "(0x20), but as five separate symbols - 0x2C, 0xE2, 0x80, 0x94, 0x20. This leads to situation when unicode 0xD180 is trimmed to 0xD1, wich is not correct unicode letter. Later this leads to error "General error: 1366 Incorrect string value: '\xD1.' for column  ..." when inserting into mysql.

`return rtrim($text, ',— ').'.';`
should be replaced with 
`return preg_replace("/([,— ]+$)/us", '', $text).'.';`

**Comment by vadimonus** [comment]
Take a look:
uk_UK: Cумно і непривітно тепер
ru_RU: На другой день Чичиков отправился на обед и вечер
ko_KR: 최석(崔晳)으로부터 최후의 편지가
kk_KZ: Арыстан баб кесенесі - көне Отырар
ka_GE: ჭეშმარიტია. ჩვენც ისე
it_IT: E PIÙ SARÀ
fr_FR: Que faisaient-elles maintenant? À
el_GR: Και δεν άκουσες το κλοπακλόπ, κλοπακλόπ, κλοπακλόπ

**Comment by ppelgrims** [comment]
Your fix doesn't pass my tests, but neither does rtrim.

```diff
     protected static function appendEnd($text)
     {
-        return rtrim($text, ',-— ').'.';
+        return preg_replace("/([,— ]+$)/us", '', $text).'.';
     }
 }
```
if I use your solution, it does work but what is getting trimmed off isn't the same?

**Comment by ppelgrims** [comment]
They do pass with this:
```diff
     protected static function appendEnd($text)
     {
-        return rtrim($text, ',-— ').'.';
+        return preg_replace("/([ \x{002C}\x{002D}\x{2013}\x{2014}]+$)/us", '', $text).'.';
     }
 }
```
But I'm uncertain what should be trimmed off: spaces, comma's, dashes, ndashes and mdashes?

**Comment by vadimonus** [comment]
I think those would be enough. I scanned all texts and they do not user \x2010, \x2012 and \x2015, so they are not needed.
You may wish to add ; and : to trimmed symbols, they are used as punctuation in many languages.
You do not need to use \x{} for , - ; : to make preg more readable, as they have code <127. Please try
`return preg_replace("/([ ,-:;\x{2013}\x{2014}]+$)/us", '', $text).'.';`
On the other hand - no matter, wich symbols you decide to trim with preg_replace, it's only important not to use rtrim. Faked text will never be perfect, and rare incorrect punctuation will not make text worse. The real problem is only when rtrim breaks unicode, causing error on inserting into database.

**Comment by akalongman** [comment]
I have the same issue as well, with the ka_GE locale. I hope it will be fixed soon

**Comment by fzaninotto** [comment]
Fixed by #1380
